### DISCUSSION TITLE
update model after loading URDF

### DISCUSSION THREAD

**Discussion by hxrkpib** [stack-of-tasks/pinocchio#1850] [Q&A] [answered]
There is a case that I want to add a payload to a quadruped robot. The payload is dynamically loaded
to body link(random mass/com/inertia).Before that, I import a model by loading URDF.
Now I have two choice:
1) update body link mass/com/inertia (treat body and payload as one link)
2) add a new link(fixed-joint) to body's child

but i don't find related API to achieve them.

**Comment by hxrkpib** [comment]
if I choice 2. add a new link(fixed-joint) to body's child, the new joint/link should can be deleted.

**Comment by jcarpent** [comment, marked as answer]
You can use `appendBodyToJoint` in the Model API

**Comment by hxrkpib** [comment]
> You can use `appendBodyToJoint` in the Model API

thanks for your reply! If I use `appendBodyToJoint`, the payload body should can be deleted for update the payload body inertia. 
in other words, the body should be add-delete-add... cycle like this to update payload body inertia(the payload mass/position maybe changed).
![screenshot-20230129-162953](https://user-images.githubusercontent.com/28836328/215314686-377298f1-914f-48d5-bec7-3283bdd25a54.png)
for example, when the robot is running, add a backpack to the body, and then remove it.

**Comment by jcarpent** [comment]
We should add the feature to remove inertia from a given joint inertia.
In the meantime, you can mimic this effect by saving the inertia of a given joint `const pinocchio::Inertia saved_inertia = model.inertias[joint_id]` and using the method `appendBodyToJoint`. When you want to remove its effect, you need to restore the saved value as `model.inertias[joint_id] = saved_inertia`. Is it fine?

**Comment by jcarpent** [comment]
@hxrkpib I will close the issue and transfer it as Q&A in the discussions.

**Comment by Gregwar** [comment]
Hello,

I have similar question so I ask it in this thread since it is related

If we update `inertias` entries in the model (e.g update a mass on the fly), do we have to run some specific computations to update related quantities (as a random example, I see there is a subtreeMasses in the data), or is it always done by relevant algorithms (like computing the mass matrix) ?
